pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -2027,6 +2028,39 @@ func getNumberOfPodInterfaces(spec *v1.VirtualMachineInstanceSpec) int {
 	return nPodInterfaces
 }
 
+func validateClientDevices(field *k8sfield.Path, clientDevices []v1.ClientDevice) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+
+	if len(clientDevices) == 0 {
+		// That's fine as this is optional
+		return nil
+	}
+
+	if len(clientDevices) > v1.UsbClientDeviceMaxNumberOf {
+		causes = append(causes, metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueNotSupported,
+			Message: fmt.Sprintf("%s does not values above %v", field.String(), v1.UsbClientDeviceMaxNumberOf),
```

Comment:
I think your error message from converter.go (`"Number of USB devices (%v) exceeds the maximum allowed (%v)"`) is better :)